pkg/pool-weighted/contracts/WeightedPoolUserData.sol

```diff
@@ -56,6 +62,11 @@ library WeightedPoolUserData {
         (, bptAmountOut) = abi.decode(self, (JoinKind, uint256));
     }
 
+    // The token must be registered before the add join, so we use the index
```

Comment:
I think it's pointing out that the argument is an index and not an address, because by the time of the join the token has already been registered and is therefore in the array. I'd remove it.